ext/handlers_signal.c

```diff
@@ -0,0 +1,75 @@
+#include "handlers_api.h"
+#include "remote_config.h"
+#include <signal.h>
+#include <php.h>
+
+/* We need to do signal blocking for the remote config signaling to not interfere with some PHP functions.
+ * See e.g. https://github.com/php/php-src/issues/16800
+ * I don't know the full problem space, so I expect there might be functions missing here, and we need to eventually expand this list.
+ */
+static void dd_handle_signal(zif_handler original_function, INTERNAL_FUNCTION_PARAMETERS) {
+    sigset_t x;
+    sigemptyset(&x);
+    sigaddset(&x, SIGVTALRM);
+    sigprocmask(SIG_BLOCK, &x, NULL);
+
+    original_function(INTERNAL_FUNCTION_PARAM_PASSTHRU);
+
+    sigprocmask(SIG_UNBLOCK, &x, NULL);
```

Comment:
If my understanding is correct, this will re-emit at least one queued signal, meaning that you may not need to call `ddtrace_check_for_new_config_now`.

> If invoking sigprocmask causes any pending signals to be unblocked, at least one of those signals is delivered to the process before sigprocmask returns. The order in which pending signals are delivered is not specified, but you can control the order explicitly by making multiple sigprocmask calls to unblock various signals one at a time. [(reference)](https://www.gnu.org/software/libc/manual/html_node/Process-Signal-Mask.html)

Reply:
Interesting, that's really nice. Manpages for non-Linux (e.g. macos) say no such thing, so I'll `#ifndef __linux__` this.